ee/tabby-webserver/src/service/db/mod.rs

```diff
@@ -20,6 +20,12 @@ lazy_static! {
         AsyncMigrations::from_directory(&MIGRATIONS_DIR).unwrap();
 }
 
+pub trait Pagination: Sized {
+    fn tab_name() -> &'static str;
+    fn select_from(source: &str, clause: &str) -> String;
```

Comment:
I feel that the trait is a bit over design in this scenario. How about having a util function like below

```rust
fn make_pagination_query(table_name: &str, field_names: [&str], limit, cursor, backwards ...) -> String {
}
```